site/app/views/ChatroomView.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace app\views;
+
+use app\entities\chat\Chatroom;
+use app\entities\chat\Message;
+use app\entities\poll\Poll;
+use app\libraries\Core;
+use app\libraries\Output;
+use app\libraries\FileUtils;
+use app\libraries\Utils;
+
+class ChatroomView extends AbstractView {
+    public function __construct(Core $core, Output $output) {
+        parent::__construct($core, $output);
+        $this->core->getOutput()->addBreadcrumb("Live Lecture Chat", $this->core->buildCourseUrl(['chat']));
+        $this->core->getOutput()->addInternalCss('chatroom.css');
+        $this->core->getOutput()->addInternalJs('chatroom.js');
+        $this->core->getOutput()->addInternalJs('websocket.js');
+    }
+
+    public function showChatPageInstructor(array $chatrooms) {
+        return $this->core->getOutput()->renderTwigTemplate("chat/ChatPageIns.twig", [
+            'csrf_token' => $this->core->getCsrfToken(),
+            'base_url' => $this->core->buildCourseUrl() . '/chat',
+            'semester' => $this->core->getConfig()->getTerm(),
+            'course' => $this->core->getConfig()->getCourse(),
+            'chatrooms' => $chatrooms
+        ]);
+    }
+
+    public function showChatPageStudent(array $chatrooms) {
+        return $this->core->getOutput()->renderTwigTemplate("chat/ChatPageStu.twig", [
+            'csrf_token' => $this->core->getCsrfToken(),
+            'base_url' => $this->core->buildCourseUrl() . '/chat',
+            'semester' => $this->core->getConfig()->getTerm(),
+            'course' => $this->core->getConfig()->getCourse(),
+            'chatrooms' => $chatrooms
+        ]);
```

Comment:
These are very similar (and the contents in the twig files are very similar too). Could we combine them and just do some conditional rendering in the twig file?

Reply:
I'll try to reduce redundancy. I wanted more clarity at this point because student's and instructor's UI could be more different.

Reply:
Although it may be tempting to have separate student and instructor UIs, we learned from the polls page that that is a bad idea.  Even though it adds additional complexity, it is important to have all of the information for a given page in one place.